src/software/thunderscope/robot_diagnostics/chicker.py

```diff
@@ -73,6 +109,90 @@ def __init__(self, proto_unix_io):
+
+        Attaches a callback to re-enable buttons after 3 seconds
+
+        :param command: Command to send. One of ChickerCommandMode.KICK or ChickerCommandMode.CHIP
+        """
+        # if button is enabled
+        if self.kick_chip_buttons_enable:
+            # send kick primitive
+            self.send_command(command)
+            self.disable_kick_chip_buttons()
+
+            # set and start timer to re-enable buttons after 3 seconds
+            QTimer.singleShot(
+                3 * MILLISECONDS_PER_SECOND, self.enable_kick_chip_buttons
+            )
+
+    def disable_kick_chip_buttons(self):
+        """
+        Disables the buttons
+        """
+        self.kick_chip_buttons_enable = False
+
+    def enable_kick_chip_buttons(self):
+        """
+        If buttons should be enabled, enables them
+        """
+        if self.no_auto_selected:
+            self.kick_chip_buttons_enable = True
+
+    def set_should_enable_buttons(self, enable):
+        """
+        Changes if buttons are clickable or not based on boolean parameter
+
+        :param enable: boolean to indicate whether buttons should be made clickable or not
+        """
+        self.no_auto_selected = enable
+
+        if enable:
+            self.enable_kick_chip_buttons()
+        else:
+            self.disable_kick_chip_buttons()
+
+    def send_command(self, command):
+        """Sends a [auto]kick or [auto]chip primitive
+
+        :param command: enum int value to indicate what primitive to send
+        :returns: None
+
+        """
+
+        # gets slider values
+        geneva_value = self.geneva_slider.value()
+
+        power_value = self.power_slider.value()
+
+        power_control = PowerControl()
+        power_control.geneva_slot = geneva_value
+
+        # sends kick, chip, autokick, or autchip primitive
+        if command == ChickerCommandMode.KICK:
+            power_control.chicker.kick_speed_m_per_s = power_value
+        elif command == ChickerCommandMode.CHIP:
+            power_control.chicker.chip_distance_meters = power_value
+        elif command == ChickerCommandMode.AUTOKICK:
+            power_control.chicker.auto_chip_or_kick.autokick_speed_m_per_s = power_value
+        elif command == ChickerCommandMode.AUTOCHIP:
+            power_control.chicker.auto_chip_or_kick.autochip_distance_meters = (
+                power_value
+            )
+
+        # sends proto
+        self.proto_unix_io.send_proto(PowerControl, power_control)
+
+        # send empty proto
+        # this is due to a bug in robot_communication where if a new PowerControl message is not sent,
+        # the previous, cached message is resent to the robot repeatedly
+        # so sending an empty message overwrites the cache and prevents spamming commands
+        # if buffer is full, blocks execution until buffer has space
+        power_control = PowerControl()
+        self.proto_unix_io.send_proto(PowerControl, power_control, True)
```

Comment:
So for powerloop in order to autokick/chip we need the breakbeam to be tripped while a autokick/chip msg is sent. Ideally it would keep track that an autokick/chip was received but this currently isn't the case. This should be fine since were sending it so often but to be clear can we only send this empty power control proto for kick/chip

Reply:
done